script.js

```diff
@@ -1,12 +1,128 @@
+      return -1;
+    }
+    if (a.name > b.name) {
+      return 1;
+    }
+    // names must be equal
+    return 0;
+  });
+
+  const showList = document.querySelector("#series-list");
+  showList.innerHTML = seriesSelector(allShows);
+
+  showList.addEventListener("change", function (event) {
+    const showId = event.target.value;
+    fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
+      .then((response) => {
+        return response.json();
+      })
+      .then((data) => {
+        makePageForEpisodes(data);
+      });
+  });
 }
 
+//content create
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  let episodes = document.querySelector(".episodeContainer");
+  episodes.innerHTML = createNewList(episodeList);
+  
+  const getInputField = document.querySelector(".search-episodes");
+
+  // Dropdown rolling Series
+  const dropDownSearchMenu = document.querySelector("#episode-list");
+  dropDownSearchMenu.addEventListener("change", function (event){
+    const episodeId = event.target.value;
+    const episodesFilteredById = episodeList.filter(
+    (episode) => episode.id === episodeId
+    );
+    episodes.innerHTML = createNewList(episodesFilteredById);
+  });
+
+  dropDownSearchMenu.innerHTML = createDropDownMenu(episodeList);
+  
+  //Search Button function and counter
+  getInputField.addEventListener("keyup", function () {
+    console.log(getInputField.value)
+      filteredEpisodes = episodeList.filter(
+      (episode) =>
+        episode.summary.toLowerCase().includes(getInputField.value.toLowerCase()) ||
+        episode.name.toLowerCase().includes(getInputField.value.toLowerCase())
+    );
+    episodes.innerHTML = createNewList(filteredEpisodes);
+    let searchCounter = document.querySelector(".counter")
+    searchCounter.textContent = `${filteredEpisodes.length} episode(s)`  
+  });
+}
+
+//Episodes Drop Down list
+function createDropDownMenu(episodeList) {
+  return episodeList
+    .map((item) => {
+      return `<option value =${item.id}>
+      S${item.season.toString().padStart(2,"0")}
+      E${item.number.toString().padStart(2, "0")} 
+      ${item.name}</option>`;
+    })
+  .join("");
+}
+
+//Episodes creation 
+function  createNewList(episodeList){
+  let searchCounter = document.querySelector(".counter")
+  searchCounter.textContent = `${episodeList.length} episode(s)`  
+return episodeList.map((item) => {
+    return `<div class="episode">
+    <h1 class="episodeHeader">${item.name} - S${item.season.toString().padStart(2, "0")}E${item.number.toString().padStart(2, "0")}</h1>
+    <img src=${item.image.medium} alt= ${item.name}>
+    ${item.summary}
+    </div>`;
+  }).join('');
+}
+
+//reset page to default
+const homeBtn = document.querySelector(".home");
+homeBtn.addEventListener("click", function () {
+  location.reload(true);
+});
+
+function seriesSelector(shows) {
```

Comment:
Is this function for the episode to be shown when the user select from the drop-down episodes lists? By the time of viewing it seems not working. Can you look in to your function again?  It can be used with similar approach like how you check search input. Really good job.